### PR TITLE
fix: Fix docs for deeply nested tables

### PR DESCRIPTION
**Pull request by hermanschaaf**
Closes https://github.com/cloudquery/plugin-sdk/issues/428

### CODE REVIEW

**Review by candiduslynx — CHANGES_REQUESTED** [review]
spaces

File: plugins/templates/table.md.go.tpl [hunk 1]
```diff
@@ -14,11 +14,11 @@ The composite primary key for this table is ({{ range $index, $pk := $.PrimaryKe
 {{- if or ($.Relations) ($.Parent) }}
 ## Relations
 {{- end }}
-{{- if $.Parent }}
+{{if $.Parent }}
```
Comment (candiduslynx) [inline plugins/templates/table.md.go.tpl:17]: what does it do?

File: plugins/templates/table.md.go.tpl [hunk 2]
```diff
@@ -14,11 +14,11 @@ The composite primary key for this table is ({{ range $index, $pk := $.PrimaryKe
...
 This table depends on [{{ $.Parent.Name }}]({{ $.Parent.Name }}.md).
 {{- end}}
 
-{{- if $.Relations }}
+{{ if $.Relations -}}
```
Comment (candiduslynx) [inline plugins/templates/table.md.go.tpl:21]: what does it do?

File: plugins/.snapshots/TestGenerateSourcePluginDocs-Markdown-test_table.md [hunk 3]
```diff
@@ -5,6 +5,8 @@ Description for test table
 The composite primary key for this table is (**id_col**, **id_col2**).
 
 ## Relations
+
+
```
Comment (candiduslynx) [inline plugins/.snapshots/TestGenerateSourcePluginDocs-Markdown-test_table.md:9]: extra empty line seems extra, maybe move it under parent presence check?

**Review by candiduslynx — CHANGES_REQUESTED** [review]

**Review by candiduslynx — APPROVED** [review]

**Review by disq — APPROVED** [review]

**Review by candiduslynx — APPROVED** [review]

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**